## Depth blob detection parameters (all are also dynamically reconfigurable)
# Parameters are listed in the order in which the corresponding
# operations are executed (except for deprecated parameters).

# image preprocessing
unknown_pixel_value: 65535
structuring_element_a: 3
structuring_element_b: 5
dilate_iterations: 0
erode_iterations: 0
erode_ignore_empty_iterations: 5

gaussianblur_size: 0
medianblur_size: 0

# filtration of detected blobs
filter_by_area: true
min_area: 100
max_area: 10000

filter_by_circularity: true
min_circularity: 0.3
max_circularity: 0.9

filter_by_orientation: true
min_angle: 0.0
max_angle: 0.4

filter_by_inertia: true
min_inertia_ratio: 0.01
max_inertia_ratio: 0.8

filter_by_convexity: true
min_convexity: 0.75
max_convexity: 1.0

filter_by_color: true
min_depth: 1000
max_depth: 40000

filter_by_known_pixels: true
min_known_pixels_ratio: 0.5

# further filtration
min_dist_between: 10.0
min_repeatability: 4

threshold_step: 500

# deprecated parameters (do not work very well/replaced by better methods)
use_threshold_width: false
threshold_width: 100
blur_empty_areas: false
